### PR TITLE
feat: set up testing server

### LINKED ISSUE #1: Build a dummy http server
## Description

In development mode, having a local dummy http server to quickly test our product is a huge plus.
A simplistic one will do. 

## Tasks

- [ ] configure a http server
- [ ] write dummy handler functions
- [ ] support any http verb (`GET`, `POST`, ...)
- [ ] write dummy handler functions

## Suggestions

Write it in Go, reason is it has the lightest dependency footprint (code and environment, eg `gofmt`, `go test`) compared to node (typescript, eslint, jest and all the jazz). 

Be super simple. `main` function may just configure the server, route the handlers and launch the server.

Use Fibonacci suite as the handler internally called function. It is quick for small numbers and exponentially grows slower. It is an excellent behavior for our usecase, easily and predictably configurable :

```txt
GET dummy/fibonacci?n=3

POST dummy/fibonacci
{ "n" : 3 }
```

### PR DESCRIPTION
**Pull request by GregoryAlbouy**
## Description

The PR includes
- Basic repository setup
- Implementation of the testing server: a single endpoint (any path, any method works) with 2 query params available:
  - `delay` (integer): delay in ms before response
  - `fib` nth fibonacci number to compute before response
- Unit tests for the handler

## Notes

- Oopsie, didn't even notice we had an issue ready, so it might not 100% follow its suggestions
- We may consider setting a max value for `fib` for safety

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,159 @@
+un:
```
Comment (GregoryAlbouy) [inline .golangci.yml]: oops

File: handle_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package main //nolint:testpackage
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/drykit-go/testx"
+	"github.com/drykit-go/testx/check"
```
Comment (moreirathomas) [inline handle_test.go:10]: 👀

File: main.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+const defaultPort = "9999"
+
+var port = flag.String("port", defaultPort, "listening port")
```
Comment (moreirathomas) [inline main.go:12]: Is there a reason it isn't within `main`?

File: handle.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+func handle(w http.ResponseWriter, r *http.Request) {
+	params := r.URL.Query()
+
+	delayMS, err := readParamInt(params, paramkeyDelay)
+	if err != nil {
+		respondError(w, 400, err)
+		return
+	}
+
+	fibIndex, err := readParamInt(params, paramkeyFib)
```
Comment (moreirathomas) [inline handle.go]: nitpick: it's not really an _index_ as in the `index`th element of a collection, and more like `n` or `x` as in `f(x)`.

File: .golangci.yml (hunk 1 above)
Comment (moreirathomas) [reply .golangci.yml]: ?

File: handle.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		return
+	}
+
+	if delayMS > 0 {
+		time.Sleep(time.Duration(delayMS) * time.Millisecond)
+	}
+
+	if fibIndex > 0 {
+		fibonacci(fibIndex) //nolint
+	}
```
Comment (moreirathomas) [inline handle.go:39]: It does not respond anything right ?
No body written could be ok (is it?) but not even a status ? 200 is the default ?

This seems kinda strange to not even declare what we're sending back to me.

Comment (GregoryAlbouy) [reply handle.go:39]: It's a 200 with empty body by default, so to me it doesn't seem necessary to explicit it

File: .golangci.yml (hunk 1 above)
Comment (GregoryAlbouy) [reply .golangci.yml]: copy-paste incident, it should be `run`

File: main.go (hunk 3 above)
Comment (GregoryAlbouy) [reply main.go:12]: None, except maybe a better visibility on what arguments the program accepts 🤷 
I believe I've seen both ways.

File: handle_test.go (hunk 2 above)
Comment (GregoryAlbouy) [reply handle_test.go:10]: This lib is amazing, I mean look how cool the unit tests look now 😍

**Comment by GregoryAlbouy** [comment]
@moreirathomas Are you fine with the current API (single handler that works with any method/path, + use of query params)?

**Comment by moreirathomas** [comment]
At the moment the server simply accepts a requests and is occupied for a predictable amount of time before responding `200 OK`.
Which is perfectly fine for where we are now regarding the runner spec (it does not do much more right now), so yes !

I predict we'll have to add more functionalities as the runner supports requests configuration.

**Review by moreirathomas — DISMISSED** [review]

**Review by moreirathomas — APPROVED** [review]

**Pull request merged**